src/components/TodoItem.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  deleteTodo: (todoId: number) => void;
+  deleteId: number[];
+  selectedaTodo: number | null | undefined;
+  onSelectedTodo: (id: number | null) => void;
+  editTodo: (todoID: number, updatedTodo: Todo) => void;
+  checkedTodo: (todo: Todo) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  deleteId,
+  selectedaTodo,
+  onSelectedTodo,
+  editTodo,
+  checkedTodo,
+}) => {
+  const deleteItem = deleteId.includes(todo.id);
+  const [renameTodo, setRenameTodo] = useState<string>(todo.title);
```

Comment:
```suggestion
  const [todoTitle, setTodoTitle] = useState<string>(todo.title);
```
`renameTodo` sounds like a function that actually renames it - wrong naming